### DISCUSSION TITLE
Not able to use prisma with mongo db

### DISCUSSION THREAD

**Discussion by VipulSinghWinit** [prisma/prisma#14180] [Q&A]
Hi! I have a mongo db schema with prisma. On running, I keep getting `Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.`  When I run prisma generate --schema='my_schema_path', it gets generated with a warning that my prisma/client and prisma are not the same version, even when they are. Then I try to run the file again and I get the same error. This error goes away when I try to import the PrismaClient from the recommended path it gives after the prisma generate command -- which is `PrismaClientInitializationError: Error parsing connection string: Database must be defined in the connection string`. even when I have `datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL_MONGO")
}`

**Comment by pimeys** [comment]
Could we have reproduction steps so we can test this out better?

**Comment by manavm1990** [comment]
In [the example from Prisma](https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/mongodb/connect-your-database-typescript-mongodb), note that the connection string used is a different format from what we would normally copy from Mongo Atlas.

The string from Mongo Atlas works for the MongoDB client, but is not the correct format for Prisma.

Instead, we need something like: `DATABASE_URL="mongodb+srv://test:test@cluster0.ns1yp.mongodb.net/myFirstDatabase"` This means specify the cluster information, remove any extraneous stuff after `.net` and replace that with `/myFirstDatabase`. 

Once we did that on a project we were working on, we no longer faced that particular issue regarding the connection string.

**Reply by vladi-stevanovic** [reply]
👋 Hello @VipulSinghWinit  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
